assets/app/view/game/buy_sell_shares.rb

```diff
@@ -138,7 +140,15 @@ def render_exchanges
             next unless company.owner == @current_entity
 
             if ability.from.include?(:ipo)
-              children.concat(render_share_exchange(@ipo_shares, company, source: @game.ipo_name(@corporation)))
+              children.concat(render_share_exchange(@ipo_shares.reject(&:president),
+                                                    company,
+                                                    source: @game.ipo_name(@corporation)))
+              if @game.exchange_for_partial_presidency? && (presidency_share = @ipo_shares.find(&:president))
+                children.concat(render_share_exchange([presidency_share],
+                                                      company,
+                                                      source: 'Presidency',
+                                                      partial_percent: 100 / presidency_share.num_shares))
```

Comment:
why is this logic in the view? shouldn't it be a part of the game?

Reply:
Do you mean the partial_percent bit or the whole thing?

Reply:
partial percent bit

Reply:
Move to game class.